salt/states/file.py

```diff
@@ -742,11 +742,17 @@ def _validate_str_list(arg):
     ensure ``arg`` is a list of strings
     '''
     if isinstance(arg, six.string_types):
-        return [arg]
+        ret = [arg]
     elif isinstance(arg, Iterable) and not isinstance(arg, Mapping):
-        return [str(item) for item in arg]
+        ret = []
+        for item in arg:
+            if isinstance(item, six.string_types):
+                ret.append(item)
+            else:
+                ret.append(str(item))
```

Comment:
It throws exceptions in Python 2 if the string is unicode and contains unicode characters which can't be decoded to ascii.

**WORKS FINE**
```python
>>> str(u'foo')
'foo'
```

**KILLS KITTENS**
```python
>>> str(u'Д')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\u0414' in position 0: ordinal not in range(128)
>>>
```